src/editor/app.zig

```diff
@@ -1,8 +1,10 @@
 const std = @import("std");
 const mach = @import("mach");
+const core = mach.core;
 const gpu = mach.gpu;
 
-pub const App = @This();
+pub const name = .editor;
+pub const App = mach.App(.{ mach.Module, @This() });
```

Comment:
Passing `@This()` into `mach.App` as a module is.. sketchy and seems very wrong. Why is this necessary?

Reply:
I get what you're doing - fine for now, but let's clean this up later.

Reply:
otherwise, init, deinit and update can't be called